apps/prairielearn/src/pages/instructorCourseAdminSettings/instructorCourseAdminSettings.ts

```diff
@@ -1,29 +1,32 @@
-//@ts-check
 import * as express from 'express';
 import * as path from 'path';
-const asyncHandler = require('express-async-handler');
+import asyncHandler = require('express-async-handler');
 import * as fs from 'fs-extra';
-const ERR = require('async-stacktrace');
+import ERR = require('async-stacktrace');
 import { CourseInfoEditor } from '../../lib/editors';
 import { logger } from '@prairielearn/logger';
 import * as error from '@prairielearn/error';
 
+import { InstructorCourseAdminSettings } from './instructorCourseAdminSettings.html';
+
 const router = express.Router();
 
 router.get(
   '/',
   asyncHandler(async (req, res) => {
+    let needToSync = false;
+    let noInfo = false;
     const pathExists = await fs.pathExists(res.locals.course.path);
     if (!pathExists) {
-      res.locals.needToSync = true;
+      needToSync = true;
     } else {
       const jsonExists = await fs.pathExists(path.join(res.locals.course.path, 'infoCourse.json'));
       if (!jsonExists) {
-        res.locals.noInfo = true;
+        noInfo = true;
       }
     }
```

Comment:
Nit, this could all be more concise.

```ts
const needToSync = !(await fs.pathExists(res.locals.course.path));
const noInfo = !(await fs.pathExists(path.join(res.locals.course.path, 'infoCourse.json')));
```

You might also consider renaming these to not be a negative, in which case you'd have the following:

```ts
const coursePathExists = await fs.pathExists(res.locals.course.path);
const courseInfoExists = await fs.pathExists(path.join(res.locals.course.path, 'infoCourse.json'))
```

You'd have to update the corresponding values passed to the HTML template and how they're actually used. I'll leave it up to you if you want to make this change.

Reply:
This is great! Updated to be more concise and use `coursePathExists` and `courseInfoExists` in [91314d4](https://github.com/PrairieLearn/PrairieLearn/pull/9406/commits/91314d4c3e75e28d2bda3cfe367f712392d9d690).